tensorflow/python/training/slot_creator.py

```diff
@@ -42,12 +42,13 @@
 from tensorflow.python.framework import ops
 from tensorflow.python.ops import array_ops
 from tensorflow.python.ops import variables
+from tensorflow.python.ops import variable_scope
 
 
 def _create_slot_var(primary, val, scope):
   """Helper function for creating a slot variable."""
 
-  slot = variables.Variable(val, name=scope, trainable=False)
+  slot = variable_scope.get_variable(scope, initializer=val, trainable=False)
```

Comment:
After reading the explanation above, I have some worries that some people will be hit by this change, e.g., they will not be able to create 2 optimizers of the same kind because they'll get reuse errors.

That's an edge case, but I think it could be avoided by putting a variable scope with default name around this get_variable call. Is this called only from the method below? If so, then let's replace the current
"ops.name_scope(socpe_name + "/")" with "variable_scope.variable_scope(None, primary.op.name)"
and then use name instead of the joint scope_name.

Does this sound reasonable?